### PR TITLE
add first iteration of the docs.

### PR DESCRIPTION
**Pull request by SantiagoTorres**
This adds the sphinx docs for pycman and pyalpm.

### CODE REVIEW

**Comment by jelly** [comment]
I like it! I would love to include either a make target and the list of dependencies to build it in our readme.

I'm not sure yet where we should/can host it. But that shouldn't be a hard problem. Checking if it builds / lints in travisci might be a nice addition.

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+# -- Project information -----------------------------------------------------
+
+project = 'pyalpm'
+copyright = '2019, Dan McGee <dan@archlinux.org>, Imanol Celaya <ilcra1989@gmail.com>, Jelle van der Waa <jelle@vdwaa.nl>, Øyvind Heggstad <mrelendig@har-ikkje.net>, Rémy Oudompheng <remy@archlinux.org>, Tasos Latsas <tlatsas2000@gmail.com>, Xyne <xyne@archlinux.ca>'
+author = 'Dan McGee <dan@archlinux.org>, Imanol Celaya <ilcra1989@gmail.com>, Jelle van der Waa <jelle@vdwaa.nl>, Øyvind Heggstad <mrelendig@har-ikkje.net>, Rémy Oudompheng <remy@archlinux.org>, Tasos Latsas <tlatsas2000@gmail.com>, Xyne <xyne@archlinux.ca>'
+
+# The short X.Y version
+version = ''
+# The full version, including alpha/beta/rc tags
+release = '0.8.4-8'
```
[the remark this answers is not in the document]
Comment (SantiagoTorres) [reply docs/conf.py]: Fixed in 9c0a1caadcfb28b1aa9b0fd3c3f4dc70a3fe93ce

File: docs/pyalpm/Database.rst [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            ( 'GROUPNAME', [package-list] ),
+            ( 'GROUPNAME2', [...]),
+            ...
+         ]
+
+
+
+   .. py:method:: get_pkg(name: string)
+
+      Retreives a package instance with the name 'name'
```
[the remark this answers is not in the document]
Comment (SantiagoTorres) [reply docs/pyalpm/Database.rst]: Fixed in ae66cb70fa49dcb4f8461002ecd0cf0824594a31

File: docs/pyalpm/Database.rst [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+     :param str name: The name of the package (e.g., 'coreutils')
+     :returns: a reference to the Package object with the name 'name' or None
+      if it doesn't exist
+
+   .. py:method:: read_group(name: string)
+
+      Retreives the list of packages that belong to the group by the name passed
```
[the remark this answers is not in the document]
Comment (SantiagoTorres) [reply docs/pyalpm/Database.rst]: Fixed in ae66cb70fa49dcb4f8461002ecd0cf0824594a31

File: docs/pyalpm/Handle.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      (i.e., an ALPM_SIG_* constant as exported in the parent module)
+     :returns: an alpm database object for this syncdb
+
+   .. py:method:: set_pkgreason(package: string, reason: pkgreason)
+
+      Sets the reason for this package installation's (e.g., explicitly or as a
+      dependency)
+
+      :param str path: the path to the cachedir to add.
+      :param str path: the path to the cachedir to add.
```
[the remark this answers is not in the document]
Comment (SantiagoTorres) [reply docs/pyalpm/Handle.rst]: fixed in 174b12e5eecf02b8e689649aad8ca886e98d4dfd

**Comment by jelly** [comment]
It looks good to me, I'm inclined to merge it. I've made a bunch of commits in my docs branch https://github.com/archlinux/pyalpm/commits/docs which fixes some warnings and contain some corrections. It seems publishing to readthedocs.io shouldn't be hard, I'm planning on reading up on their documentation.

**Comment by jelly** [comment]
For readthedocs we need, to just connect our account and setup a hook.

https://docs.readthedocs.io/en/stable/webhooks.html

**Pull request merged**